flytekit/image_spec/image_spec.py

```diff
@@ -103,8 +104,11 @@ def exist(self) -> bool:
         """
         Check if the image exists in the registry.
         """
-        import docker
-        from docker.errors import APIError, ImageNotFound
+
+        try:
+            import docker
```

Comment:
@thomasjpfan I'm thinking of using `subprocess.run("docker ...")` to check if the image exists. Is there any concern about using a subprocess here?

Reply:
If we can use a subprocess, people don't even need to install the docker package here.

Reply:
I am okay with using `subprocess` here.

Reply:
Will do it, thank you!